pkg/flow/flow.go

```diff
@@ -259,14 +277,14 @@ func (f *Flow) Run(ctx context.Context) {
 // file. Components in the graph will be marked as unhealthy if there was an
 // error encountered during Load.
 //
-// The controller will only start running components after Load is called once
-// without any configuration errors.
-func (f *Flow) LoadFile(file *File, args map[string]any) error {
+// If strict is true, the controller will only start running components after
+// Load is called once without any configuration errors.
+func (f *Flow) LoadFile(file *File, args map[string]any, strict bool) error {
 	f.loadMut.Lock()
 	defer f.loadMut.Unlock()
 
 	diags := f.loader.Apply(args, file.Components, file.ConfigBlocks)
-	if !f.loadedOnce.Load() && diags.HasErrors() {
+	if strict && !f.loadedOnce.Load() && diags.HasErrors() {
```

Comment:
I don't adding an argument to LoadFile is required; the caller should decide whether or not an error from LoadFile is fatal before running the controller. 

The rest of this code is still necessary though.